src/components/pages/AllMoviesPage.vue

```diff
@@ -0,0 +1,116 @@
+<script>
+import { defineComponent } from 'vue';
+import { mapState } from "pinia";
+import movies from '@/stores/moviesStore';
+import BreadCrumbs from '@/components/common/BreadCrumbs.vue';
+import SectionTitle from '@/components/common/Section/SectionTitle.vue';
+import AppSelect from '@/components/common/App/AppSelect.vue';
+import MovieCard from '@/components/views/MovieCard.vue';
+import AppInput from '@/components/common/App/AppInput.vue';
+import AppLabel from '@/components/common/App/AppLabel.vue';
+import TheContainer from '@/components/common/TheContainer.vue';
+
+export default defineComponent({
+  components: { BreadCrumbs, SectionTitle, AppSelect, MovieCard, AppInput, AppLabel, TheContainer },
+  data() {
+    return {
+      optionSelected: 'All categories',
+      inputContent: '',
+    }
+  },
+  computed: {
+    ...mapState(movies, ["movies"]),
+    moviesCategories() {
+      const allCategories = this.movies.map(movie => movie.genre.name)
+      return new Set(allCategories)
+    },
+    moviesFilteredByCategory() {
+      if (this.optionSelected !== 'All categories') {
+        return this.movies.filter(movie => this.optionSelected === movie.genre.name)
+      } else {
+        return this.movies
+      }
+    },
+    moviesFilteredByCategoryAndSearch() {
+      if (this.inputContent !== '') {
+        return this.moviesFilteredByCategory.filter(movie => movie.title.toLowerCase().includes(this.inputContent.toLowerCase()))
+      } else {
+        return this.moviesFilteredByCategory
+      }
+    },
+  }
+});
+</script>
+
+<template>
+  <TheContainer>
+    <div class="all-movies">
+      <BreadCrumbs>Movies</BreadCrumbs>
+      <SectionTitle size="large">All the movies</SectionTitle>
+      <div class="all-movies__filters">
+        <div class="all-movies__filters--select">
+          <AppLabel>Search</AppLabel>
+          <AppInput
+            placeholder="What are you looking for?"
+            v-model="inputContent"
+          />
+        </div>
+        <div class="all-movies__filters--select">
+          <AppLabel>Category</AppLabel>
+          <AppSelect
+            :options="moviesCategories"
+            v-model="optionSelected"
+            optionStart="All categories"
+            label-text="category"
+          ></AppSelect>
+        </div>
+      </div>
+      <div class="all-movies__cards-list">
+        <MovieCard
+          v-for="movie of moviesFilteredByCategoryAndSearch"
+          :key="movie.id"
+          :movie="movie"
+        />
+      </div>
```

Comment:
A suggestion - how about we add a small improvement and show some information when we do not have any movies to display - I know it's not covered by designs 😅

This way if someone enters a phrase or a movie name that's not found (or checks up on day with no seances there) - they see some kind of message or information instead of... well, nothing 👀

We could also add this in to the seances list - because if we filter by a certain movie and date - we have cases where there are no seances shown too.

Smells like a nice reusable component to have this message shown in a smiliar fashion in multiple places 👀

Reply:
Yeah, I've actually thought about it but then got distracted by other issues -- will add it soon! 👍